### PR TITLE
NXP-21349 [on 8.10-SNAPSHOT] Split nuxeo-tree-snapshot in core and jsf parts

### PR DESCRIPTION
**Pull request by egiuly-nuxeo**
https://jira.nuxeo.com/browse/NXP-21349

I split `nuxeo-tree-snapshot` into `nuxeo-tree-snapshot-core` and `nuxeo-tree-snapshot-jsf`.

In `nuxeo-tree-snapshot-jsf` I put :
- `src/main/java/org/nuxeo/snapshot/bean/VFolderActionBean.java`,
- `src/main/resources/web/`,
- `src/main/resources/OSGI-INF/` `actions-contrib.xml`,`contentviews-contrib.xml`,`deployment-fragment.xml`.

In `nuxeo-tree-snapshot-core` I put the rest and all the tests.

I changed the `groupId` from `org.nuxeo.features` to `org.nuxeo.snapshot` because the package was called `org.nuxeo.snapshot`.

This PR is on 8.10-SNAPSHOT, so I put the versions:
`nuxeo-addons-parent` => 8.10-SNAPSHOT
`nuxeo-tree-snapshot` => 8.10.1-SNAPSHOT
`nuxeo-tree-snapshot-core` => 8.10.1-SNAPSHOT
`nuxeo-tree-snapshot-jsf` => 8.10.1-SNAPSHOT

### CODE REVIEW

File: nuxeo-tree-snapshot-jsf/src/main/resources/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 1
+Bundle-Name: Nuxeo ECM Snapshot Manager
+Bundle-SymbolicName: org.nuxeo.snapshot;singleton:=true
+Bundle-Version: 1.0.0
+Nuxeo-Component: OSGI-INF/publisher-contrib.xml, OSGI-INF/contentviews-contrib.xml
```
Comment (efge) [inline nuxeo-tree-snapshot-jsf/src/main/resources/META-INF/MANIFEST.MF]: It's not `publisher-contrib.xml` but `actions-contrib.xml`.

File: nuxeo-tree-snapshot-jsf/src/main/resources/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 1
+Bundle-Name: Nuxeo ECM Snapshot Manager
+Bundle-SymbolicName: org.nuxeo.snapshot;singleton:=true
```
Comment (efge) [inline nuxeo-tree-snapshot-jsf/src/main/resources/META-INF/MANIFEST.MF]: You must change the symbolic name to be different than the core one. For instance `org.nuxeo.snapshot.jsf`.

File: nuxeo-tree-snapshot-jsf/src/main/resources/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 1
+Bundle-Name: Nuxeo ECM Snapshot Manager
```
Comment (efge) [inline nuxeo-tree-snapshot-jsf/src/main/resources/META-INF/MANIFEST.MF]: Nuxeo ECM Snapshot Manager (JSF)

File: nuxeo-tree-snapshot-core/pom.xml [hunk 4]
```diff
@@ -0,0 +1,175 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.nuxeo.snapshot</groupId>
```
Comment (efge) [inline nuxeo-tree-snapshot-core/pom.xml]: Use `org.nuxeo.ecm.platform` here, like most addons, we don't want to multiply the group ids and especially have them be specific to the addon.

File: pom.xml [hunk 5]
```diff
@@ -7,194 +7,15 @@
     <version>8.10-SNAPSHOT</version>
   </parent>
 
-  <groupId>org.nuxeo.features</groupId>
+  <groupId>org.nuxeo.snapshot</groupId>
```
Comment (efge) [inline pom.xml]: `org.nuxeo.ecm.platform`

File: nuxeo-tree-snapshot-core/pom.xml [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    <artifactId>nuxeo-tree-snapshot</artifactId>
+    <version>8.10.1-SNAPSHOT</version>
+  </parent>
+
+  <groupId>org.nuxeo.snapshot</groupId>
```
Comment (efge) [inline nuxeo-tree-snapshot-core/pom.xml]: `org.nuxeo.ecm.platform`

File: nuxeo-tree-snapshot-jsf/pom.xml [hunk 7]
```diff
@@ -0,0 +1,49 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.nuxeo.snapshot</groupId>
```
Comment (efge) [inline nuxeo-tree-snapshot-jsf/pom.xml]: `org.nuxeo.ecm.platform`

File: nuxeo-tree-snapshot-jsf/pom.xml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    <artifactId>nuxeo-tree-snapshot</artifactId>
+    <version>8.10.1-SNAPSHOT</version>
+  </parent>
+
+  <groupId>org.nuxeo.snapshot</groupId>
```
Comment (efge) [inline nuxeo-tree-snapshot-jsf/pom.xml]: `org.nuxeo.ecm.platform`

File: nuxeo-tree-snapshot-jsf/pom.xml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  <artifactId>nuxeo-tree-snapshot-jsf</artifactId>
+  <name>Nuxeo Tree Snapshot JSF</name>
+
+  <dependencies>
+  	<dependency>
+      <groupId>org.nuxeo.ecm.core</groupId>
+      <artifactId>nuxeo-core-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.snapshot</groupId>
```
Comment (efge) [inline nuxeo-tree-snapshot-jsf/pom.xml]: `org.nuxeo.ecm.platform`

**Review by efge — CHANGES_REQUESTED** [review]

File: nuxeo-tree-snapshot-core/pom.xml [hunk 10]
```diff
@@ -0,0 +1,175 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.nuxeo.snapshot</groupId>
+    <artifactId>nuxeo-tree-snapshot</artifactId>
+    <version>8.10.1-SNAPSHOT</version>
```
Comment (nelsonsilva) [inline nuxeo-tree-snapshot-core/pom.xml:7]: nuxeo-tree-snapshot parent in`8.10-SNAPSHOT` is not using this `8.10.1-SNAPSHOT` version

**Review by nelsonsilva — CHANGES_REQUESTED** [review]
Seems this has been released for 8.10 already, no?

**Review by efge — APPROVED** [review]

**Comment by efge** [comment]
👍

**Review by nelsonsilva — APPROVED** [review]

**Comment by nelsonsilva** [comment]
:+1:

**Pull request closed without merging**